### PR TITLE
Upgrade sqlalchemy to 1.4

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR migrates housekeeper to SQLAlchemy 1.4. Part of https://github.com/Clinical-Genomics/cg/issues/2497.

### Fixed
- Migrate SQLAlchemy to 1.4

Blocked by upgrading cg to SQLAlchemy 1.4 as well.

EDIT:
To be able to start migrating cg to 1.4, the requirement for sqlalchemy is set to <=1.4 and this PR needs to be merged.

This [version](https://semver.org/) is a:
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: housekeeper/cli/init.py [hunk 1]
```diff
@@ -24,4 +28,4 @@ def init(context, reset, force):
         context.abort()
 
     store.create_all()
-    LOG.info("Success! New tables: %s", ", ".join(store.engine.table_names()))
+    LOG.info("Success! New tables: %s", ", ".join(inspector.get_table_names()))
```
Comment (diitaz93) [inline housekeeper/cli/init.py:31]: use formatted strings

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by seallard** [comment]
Deployed and tested in stage

**Pull request merged**